sqlserver/check.py

```diff
@@ -101,12 +112,58 @@ def __init__(self, name, init_config, agentConfig, instances=None):
         self.custom_metrics = init_config.get('custom_metrics', [])
         for instance in instances:
             try:
-                with self.open_managed_db_connections(instance):
-                    self._make_metric_list_to_collect(instance, self.custom_metrics)
+                instance_key = self._conn_key(instance, self.DEFAULT_DB_KEY)
+                self.do_check[instance_key] = False
+
+                # check to see if the database exists before we try any connections to it
+                with self.open_managed_db_connections(instance, None, db_name=self.DEFAULT_DATABASE):
+                    db_exists, context = self._check_db_exists(instance)
+
+                if db_exists:
+                    self.do_check[instance_key] = True
+                    if instance.get('stored_procedure') is None:
+                        with self.open_managed_db_connections(instance, self.DEFAULT_DB_KEY):
+                            self._make_metric_list_to_collect(instance, self.custom_metrics)
+                else:
+                    # How much do we care that the DB doesn't exist?
+                    ignore = instance.get('ignore_missing_database')
```

Comment:
The build is failing because `_is_affirmative` is unknown. Should this be `self.` or something else?